lib/fromPgn.js

```diff
@@ -800,7 +800,7 @@ fieldTypeReaders["String with start/stop byte"] = (pgn, field, bs) => {
       var c = bs.readUint8()
       buf.writeUInt8(c, idx)
     }
-    return buf.toString('ascii', 0, idx)
+    return buf.toString('ascii', 0, idx).replace(/\0.*$/g,'')
```

Comment:
The regex is not correct for matching a trailing null.

<img width="431" alt="image" src="https://user-images.githubusercontent.com/1049678/227890842-4bf84473-7446-48ae-a482-f97da3b85afa.png">

```
> r = /\0.*$/
/\0.*$/
> s = '\0more'
'\x00more'
> !!s.match(r)
true
```

Even if it were: are there no legit cases where the last byte would not be null?

Reply:
In B&G waypoints the current function includes the \u0000 at the end of the route/waypoint name.
The regex does remove it and works when replaying the pgn.

I noticed trim() functions in other string functions, so this isn't that different.

Probably using an approach like 'ASCII text' is better using a variety of stopbytes, but I didn't want to rewrite the whole function as I don't know how to test many cases.

Reply:
Here's the pgn I using to check:
`2023-03-26T20:00:05.233Z,7,129285,28,255,8,c0,3b,09,00,02,00,ff,ff
2023-03-26T20:00:05.234Z,7,129285,28,255,8,c1,ff,ff,e0,0b,01,42,41
2023-03-26T20:00:05.253Z,7,129285,28,255,8,c2,52,42,41,44,4f,53,00
2023-03-26T20:00:05.254Z,7,129285,28,255,8,c3,16,00,00,09,01,52,70
2023-03-26T20:00:05.254Z,7,129285,28,255,8,c4,74,30,31,31,00,d9,d5
2023-03-26T20:00:05.255Z,7,129285,28,255,8,c5,12,05,2e,c9,75,df,01
2023-03-26T20:00:05.256Z,7,129285,28,255,8,c6,00,09,01,52,70,74,30
2023-03-26T20:00:05.257Z,7,129285,28,255,8,c7,31,33,00,55,96,c0,07
2023-03-26T20:00:05.257Z,7,129285,28,255,8,c8,e8,98,74,dc,ff,ff,ff
`